### PR TITLE
hw-10.1-solution

### PR DESCRIPTION
**Pull request by SviatoslavExpert**

### CODE REVIEW

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 1]
```diff
@@ -17,7 +23,31 @@
...
 public class TreeSetCharacters {
-    public String getUniqueCharacters(String fileName) {
-        return null;
+
+    private static final int CHARS_NUMBER = 5;
+
+    public String getUniqueCharacters(String fileName) throws IOException {
+        String resultString = "";
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            Set<Character> set = new TreeSet<>();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: suggestion: set rename to characters

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 2]
```diff
@@ -17,7 +23,31 @@
...
+            while (bufferedReader.ready()) {
+                String letters = bufferedReader.readLine().toLowerCase()
+                        .replaceAll("[^a-zA-Z]", "");
+                for (int i = 0; i < letters.length(); i++) {
+                    set.add(letters.charAt(i));
+                }
+            }
+            int size = set.size();
+            if (size > CHARS_NUMBER) {
+                resultString = set.toString().replaceAll("[^a-zA-Z]", "")
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: should we use replaceAll("[^a-zA-Z]", "") here?

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 3]
```diff
@@ -17,7 +23,31 @@
...
+                        .substring(0, CHARS_NUMBER);
+                return resultString;
+            }
+            resultString = set.toString().replaceAll("[^a-zA-Z]", "");
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: the same)

File: src/main/java/core/basesyntax/TreeSetCharacters.java [hunk 4]
```diff
@@ -17,7 +23,31 @@
...
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            Set<Character> set = new TreeSet<>();
+            while (bufferedReader.ready()) {
+                String letters = bufferedReader.readLine().toLowerCase()
+                        .replaceAll("[^a-zA-Z]", "");
+                for (int i = 0; i < letters.length(); i++) {
+                    set.add(letters.charAt(i));
+                }
+            }
+            int size = set.size();
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/TreeSetCharacters.java]: we can simplify:
resultString = set.toString().replaceAll("[^a-zA-Z]", "");
            if (set.size() > CHARS_NUMBER) {
return resultString.substring(0, CHARS_NUMBER);
            }
            return resultString;

File: src/main/java/core/basesyntax/TreeSetCharacters.java (hunk 1 above)
Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: Fixed.

File: src/main/java/core/basesyntax/TreeSetCharacters.java (hunk 2 above)
Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: Fixed.

File: src/main/java/core/basesyntax/TreeSetCharacters.java (hunk 3 above)
Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: Fixed.

File: src/main/java/core/basesyntax/TreeSetCharacters.java (hunk 4 above)
Comment (SviatoslavExpert) [reply src/main/java/core/basesyntax/TreeSetCharacters.java]: Fixed.

**Review by Sofasmile — APPROVED** [review]

**Pull request closed without merging**